Entities/Industry/Drill/Drill.as

```diff
@@ -388,9 +391,44 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
 {
 	this.getCurrentScript().runFlags &= ~Script::tick_not_sleeping;
+	this.set_s32("showHeatTo", getPlayerIndex(attached.getPlayer()));
+}
+
+void onDetach(CBlob@ this, CBlob@ detached, AttachmentPoint @attachedPoint)
+{
+	this.set_s32("showHeatTo", -1);
 }
 
 void onThisAddToInventory(CBlob@ this, CBlob@ blob)
 {
 	this.getSprite().SetEmitSoundPaused(true);
 }
+
+void onRender(CSprite@ this)
+{
+	CBlob@ blob = this.getBlob();
+	CPlayer@ holder = getPlayer(blob.get_s32("showHeatTo"));
+	
+	Vec2f mousePos = getControls().getMouseWorldPos();
+	Vec2f blobPos = blob.getPosition();
+
+	if (getLocalPlayer() is null || getLocalPlayer().getBlob() is null)
+		return;
+
+	bool inRange = (blobPos - getLocalPlayer().getBlob().getPosition()).getLength() < max_heatbar_view_range;
```

Comment:
No need to repeatedly get the local player (on lines 415 and 418). Get it once and store it in a variable, same with the local player's blob.

Reply:
I've updated my ```onRender``` function make more efficient use of variables.